api/utils/sshutils/chconn.go

```diff
@@ -125,11 +125,11 @@ func (c *ChConn) RemoteAddr() net.Addr {
 // Read reads from the channel.
 func (c *ChConn) Read(data []byte) (int, error) {
 	n, err := c.reader.Read(data)
-	// A lot of code relies on "use of closed network connection" error to
+	// A lot of code relies on net.ErrClosed or "use of closed network connection" error to
 	// gracefully handle terminated connections so convert the closed pipe
 	// error to it.
 	if err != nil && err == io.ErrClosedPipe {
-		return n, trace.ConnectionProblem(err, constants.UseOfClosedNetworkConnection)
+		return n, trace.Wrap(net.ErrClosed, constants.UseOfClosedNetworkConnection)
```

Comment:
Are we abandoning the use of `trace.ConnectionProblem`? If so, we may want to drop it from the `trace` library or deprecate it. Notably, `trace.ConvertSystemError` may still produce it.

Reply:
I'm going to restore some of those `trace.ConnectionProblem`s but making sure that they contain a `net.ErrClosed` (or something that contains it), as the check for `errors.Is(err, net.ErrClosed)` would succeed anyway.

I'm not sure of the semantics of `ConnectionProblem`, however: is an error on `(net.Listener).Accept()` on a closed listener really a *connection* problem?

Reply:
Turned some of these into `ConnectionProblem(net.ErrClosed, "message")` and left alone `chconn.go` - it's handled by the check on the error message, anyway.